test/providers/provider_level_zero.cpp

```diff
@@ -326,21 +326,66 @@ struct umfLevelZeroProviderTest : umf_test::test {
         ASSERT_EQ(ze_result, ZE_RESULT_SUCCESS);
     }
 
-    ze_driver_handle_t hDriver;
-    ze_device_handle_t hDevice;
-    ze_context_handle_t hContext;
-    void *zeDlHandle;
+  private:
+    ze_device_handle_t hDevice_;
+    ze_context_handle_t hContext_;
+};
+
+using LevelZeroProviderTestParams =
+    std::tuple<level_zero_memory_provider_params_t, MemoryAccessor *>;
+
+struct umfLevelZeroProviderTest
+    : umf_test::test,
+      ::testing::WithParamInterface<LevelZeroProviderTestParams> {
+
+    void SetUp() override {
+        test::SetUp();
+
+        auto [l0_params, accessor] = this->GetParam();
+        params = l0_params;
+        hDevice = (ze_device_handle_t)params.level_zero_device_handle;
+        hContext = (ze_context_handle_t)params.level_zero_context_handle;
+
+        if (nullptr == hDevice) {
+            GTEST_SKIP() << "Test skipped, no GPU devices found";
+        }
+
+        switch (params.memory_type) {
+        case UMF_MEMORY_TYPE_DEVICE:
+            zeMemoryTypeExpected = ZE_MEMORY_TYPE_DEVICE;
+            break;
+        case UMF_MEMORY_TYPE_SHARED:
+            zeMemoryTypeExpected = ZE_MEMORY_TYPE_SHARED;
+            break;
+        case UMF_MEMORY_TYPE_HOST:
+            zeMemoryTypeExpected = ZE_MEMORY_TYPE_HOST;
+            break;
+        default:
+            GTEST_SKIP() << "Test skipped, unsupported memory type";
```

Comment:
fail here as well...?

Reply:
Done. I have removed `default` case and added `ASSERT_NE(zeMemoryTypeExpected, ZE_MEMORY_TYPE_UNKNOWN);` after `switch` block.